### PR TITLE
chore: scafford part 0

### PR DESCRIPTION
**Pull request by tokers**
Signed-off-by: tokers <zchao1995@gmail.com>

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+package cloud
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
```
Comment (Hazel6869) [inline http.go:23]: import sort

File: options.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+// limitations under the License.
+
+package cloud
+
+import "time"
+
+var (
+	// DefaultOptions contains the default settings for all the options.
+	DefaultOptions = &Options{
+		ServerAddr:            "https://api.api7.cloud",
```
Comment (Baoyuantop) [inline options.go:22]: Do we need to include `api/v1`?

File: http.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+	if err != nil {
+		return errors.Wrap(err, "encode http request body")
+	}
+
+	req, err := http.NewRequest(http.MethodPut, u.String(), bytes.NewReader(data))
+	if err != nil {
+		return errors.Wrap(err, "construct http request")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
```
Comment (Baoyuantop) [inline http.go:159]: Is it possible to handle it in sendRequest uniformly?

**Review by oil-oil — APPROVED** [review]

File: options.go (hunk 2 above)
Comment (tokers) [reply options.go:22]: > Do we need to include api/v1?

It's just an address, without the path prefix, the path prefix `/api/v1` is the API version and the Go SDK will be consistent with it. We may have a V2 Go SDK and in such a case, will use `/api/v2`, so it's not suitable to put the path prefix to `ServerAddr`.

File: http.go (hunk 3 above)
Comment (tokers) [reply http.go:159]: `sendRequest` doesn't know the type of request body.

File: options.go (hunk 2 above)
Comment (tokers) [reply options.go:22]: > should this store be in env?

SDK should not consider this. It's the responsibility of caller of SDK .

File: http.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+type httpClientImpl struct {
+	url    *url.URL
+	token  *AccessToken
+	client *http.Client
+}
+
+func constructHTTPClient(opts *Options, token *AccessToken) (httpClient, error) {
+	u, err := url.Parse(opts.ServerAddr)
```
[the remark this answers is not in the document]
Comment (tokers) [reply http.go]: Fixed.

File: http.go (hunk 1 above)
Comment (tokers) [reply http.go]: Fixed.

**Review by Hazel6869 — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Pull request merged**